torch/ao/quantization/fx/_model_report/detector.py

```diff
@@ -529,8 +555,37 @@ def determine_observer_insert_points(self, prepared_fx_model: GraphModule) -> Di
             key "observer_args" -> The arguments that are meant to be passed into the observer
         """
 
+        # observer for this detector is ModelReportObserver
+        obs_ctr = ModelReportObserver
+
+        # return dict
+        obs_fqn_to_info: Dict[str, Dict[str, Any]] = {}
+
+        for fqn, module in prepared_fx_model.named_modules():
+            # check to see if module is of a supported type
+            is_supported_type: bool = (
+                sum(list(map(lambda x: type(module) is x, self.SUPPORTED_MODULES))) > 0
+            )
+
+            if is_supported_type:
+                # if it's a supported type, we want to get node and add observer insert locations
+
+                targeted_node = self._get_targeting_node(prepared_fx_model, fqn)
+
+                # add entry for pre-observer
+                pre_obs_fqn = fqn + "." + self.DEFAULT_PRE_OBSERVER_NAME
+
+                obs_fqn_to_info[pre_obs_fqn] = {
+                    "target_node": targeted_node,
+                    "insert_observer": obs_ctr,
+                    "insert_post": False,
+                    "observer_args": targeted_node.args,
+                }
```

Comment:
This is starting to look a lot like [L277-L331](https://github.com/pytorch/pytorch/blob/21d51e9740c43038792cfd3487342bf3d7ded964/torch/ao/quantization/fx/_model_report/detector.py#L277-L331). I wonder if we should factor it in a common helper method that takes in the following args to reduce duplicate code:
```
supported_modules: List[...],
pre_observer_name: str,
post_observer_name: Optional[str]
```

Reply:
i think there is more clarity in the next commit in the stack, https://github.com/pytorch/pytorch/pull/80191, but this does diverge since the ModelReport Observer is instantiated in different ways within this loop depending on what the observer is, so I thought there was enough difference in implementation that it warranted being separate. That being said, I will definitely keep this in mind and focus more on refactoring where appropriate as I continue working on this.